File: Problem-040/solution.js
function champernownesConstant(n) {
    let str = "";
    let num = 1;

    // Keep adding numbers until we have at least n digits
    while (str.length < n) {
        str += num.toString();
        num++;
    }

    let product = 1;

    // Positions: 1, 10, 100, ..., n
    for (let position = 1; position <= n; position *= 10) {
        product *= Number(str[position - 1]);
    }

    return product;
}

console.log(champernownesConstant(1000000)); // 210

File: Problem-040/solution.py
def champernownes_constant(n):
    string = ""
    num = 1

    # Build the constant
    while len(string) < n:
        string += str(num)
        num += 1

    product = 1

    # Get positions 1, 10, 100, ..., n
    position = 1

    while position <= n:
        product *= int(string[position - 1])
        position *= 10

    return product


print(champernownes_constant(1_000_000))  # 210
